app/services/bill-runs/two-part-tariff/fetch-bill-run-licences.service.js

```diff
@@ -45,22 +63,42 @@ async function _fetchBillRun (id) {
     })
 }
 
-async function _fetchBillRunLicences (id, licenceHolder, licenceStatus) {
+async function _fetchBillRunLicences (id, filterIssues, filterLicenceHolder, filterLicenceStatus) {
   const reviewLicenceQuery = ReviewLicenceModel.query()
     .where('billRunId', id)
     .orderBy('status', 'desc')
 
-  if (licenceHolder) {
-    reviewLicenceQuery.whereILike('licenceHolder', `%${licenceHolder}%`)
-  }
-
-  if (licenceStatus) {
-    reviewLicenceQuery.where('status', licenceStatus)
-  }
+  _applyFilters(reviewLicenceQuery, filterIssues, filterLicenceHolder, filterLicenceStatus)
 
   return reviewLicenceQuery
 }
 
+function _filterIssues (filterIssues, reviewLicenceQuery) {
+  // if only a single issue is checked in the filter then a string is returned, otherwise it is an array
+  if (typeof filterIssues === 'string') {
+    const lookupIssue = twoPartTariffReviewIssues[filterIssues]
+    reviewLicenceQuery.whereLike('issues', `%${lookupIssue}%`)
+  } else {
+    // if we have got here then `issues` must be an array containing at least 2 records
+    const lookupIssues = filterIssues.map((filterIssue) => {
+      return twoPartTariffReviewIssues[filterIssue]
+    })
+
+    // the number of issues to check for in the where clause will vary depending on the number of issues checked. But
+    // there will always be at least 2
+    reviewLicenceQuery.where((builder) => {
```

Comment:
Can this be condensed like so?
```suggestion
    reviewLicenceQuery.where((builder) => {
      builder
        .whereLike('issues', `%${lookupIssues[0]}%`)
      for (let i = 1; i < lookupIssues.length; i++) {
        builder.orWhereLike('issues', `%${lookupIssues[i]}%`)
      }
    })
```

Reply:
Yes, I'll do that